theano/tensor/nnet/abstract_conv.py

```diff
@@ -430,6 +441,35 @@ def do_constant_folding(self, node):
         # This may change in the future.
         return False
 
+    def corr2d(self, img, kern, mode="valid"):
+        """
+        Basic slow python implementatation for DebugMode
+        """
+
+        if not imported_scipy_signal:
+            raise theano.gof.utils.MethodNotDefined(
+                "c_headers", type(self), self.__class__.__name__,
+                "Need the python package for scipy.signal to be installed "
+                "for the python implementation. You can use the C"
+                " implementation instead.")
```

Comment:
I don't think that is the right exception type to raise, and the message should be updated (is there a C implementation?)